src/zenml/integrations/gcp/model_registries/vertex_model_registry.py

```diff
@@ -0,0 +1,308 @@
+        except Exception as e:
+            raise RuntimeError(f"Failed to delete model: {str(e)}")
+
+    def update_model(
+        self,
+        name: str,
+        description: Optional[str] = None,
+        metadata: Optional[Dict[str, str]] = None,
+        remove_metadata: Optional[List[str]] = None,
+    ) -> RegisteredModel:
+        """Update a model in the Vertex AI model registry."""
+        raise NotImplementedError(
+            "Vertex AI does not support updating models, you can only update model versions, skipping model registration..."
+        )
+
+    def get_model(self, name: str) -> RegisteredModel:
+        """Get a model from the Vertex AI model registry."""
+        try:
+            model = aiplatform.Model(model_name=name)
+            return RegisteredModel(
+                name=model.name,
+                description=model.description,
+                metadata=model.labels,
+            )
+        except Exception as e:
+            raise RuntimeError(f"Failed to get model: {str(e)}")
+
+    def list_models(
+        self,
+        name: Optional[str] = None,
+        metadata: Optional[Dict[str, str]] = None,
+    ) -> List[RegisteredModel]:
+        """List models in the Vertex AI model registry."""
+        self.setup_aiplatform()
+        filter_expr = 'labels.managed_by="zenml"'
+        if name:
+            filter_expr = filter_expr + f' AND  display_name="{name}"'
+        if metadata:
+            for key, value in metadata.items():
+                filter_expr = filter_expr + f' AND  labels.{key}="{value}"'
+        try:
+            models = aiplatform.Model.list(filter=filter_expr)
+            return [
+                RegisteredModel(
+                    name=model.display_name,
+                    description=model.description,
+                    metadata=model.labels,
+                )
+                for model in models
+            ]
+        except Exception as e:
+            raise RuntimeError(f"Failed to list models: {str(e)}")
+
+    def register_model_version(
+        self,
+        name: str,
+        version: Optional[str] = None,
+        model_source_uri: Optional[str] = None,
+        description: Optional[str] = None,
+        metadata: Optional[ModelRegistryModelMetadata] = None,
+        **kwargs: Any,
+    ) -> RegistryModelVersion:
+        """Register a model version to the Vertex AI model registry."""
+        self.setup_aiplatform()
+        metadata_dict = metadata.model_dump() if metadata else {}
+        serving_container_image_uri = metadata_dict.get(
+            "serving_container_image_uri",
+            None
+            or "europe-docker.pkg.dev/vertex-ai/prediction/sklearn-cpu.1-3:latest",
+        )
+        is_default_version = metadata_dict.get("is_default_version", False)
+        try:
+            version_info = aiplatform.Model.upload(
+                artifact_uri=model_source_uri,
+                display_name=f"{name}_{version}",
```

Comment:
I remember one other important thing related to multi-tenancy: imagine that multiple zenml tenants/servers might want to register models in the same region. If they want to use the same model name, there might be a name clash between them (I don't know if you can have two Vertex AI models with the same display name). To avoid this:

* you could use the ZenML server UUID as a label to differentiate, and filter results only by that label
* if the display name also has to be unique, you could prepend the server UUID to the display name, or at least part of it